### PR TITLE
[improve][broker] Make some operation SubscribeRate methods in Namespaces async

### PR DESCRIPTION
**Pull request by gaoyanliang**
### Motivation

See PIP https://github.com/apache/pulsar/issues/14365 and change tracker https://github.com/apache/pulsar/issues/15043.

Make NamespacesBase deleteSubscribeRate / setSubscribeRate / getSubscribeRate methods to pure async.

### Documentation

- [x] `no-need-doc`

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java [hunk 1]
```diff
@@ -1373,43 +1373,41 @@ protected DispatchRate internalGetSubscriptionDispatchRate() {
...
+        return validateSuperUserAccessAsync().thenAccept(__ -> {
             updatePolicies(namespaceName, policies -> {
                 policies.clusterSubscribeRate.put(pulsar().getConfiguration().getClusterName(), subscribeRate);
                 return policies;
             });
             log.info("[{}] Successfully updated the subscribeRate for cluster on namespace {}", clientAppId(),
                     namespaceName);
-        } catch (Exception e) {
+        }).exceptionally(ex -> {
             log.error("[{}] Failed to update the subscribeRate for cluster on namespace {}", clientAppId(),
```
[the remark this answers is not in the document]
Comment (shibd) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:1386]: And keep one error log print

**Comment by gaoyanliang** [comment]
@shibd  All suggested fixed. PTAL.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java [hunk 2]
```diff
@@ -1373,43 +1373,33 @@ protected DispatchRate internalGetSubscriptionDispatchRate() {
         return policies.subscriptionDispatchRate.get(pulsar().getConfiguration().getClusterName());
     }
 
-    protected void internalSetSubscribeRate(SubscribeRate subscribeRate) {
-        validateSuperUserAccess();
+    protected CompletableFuture<Void> internalSetSubscribeRateAsync(SubscribeRate subscribeRate) {
         log.info("[{}] Set namespace subscribe-rate {}/{}", clientAppId(), namespaceName, subscribeRate);
-        try {
+        return validateSuperUserAccessAsync().thenAccept(__ -> {
             updatePolicies(namespaceName, policies -> {
```
Comment (mattisonchao) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:1379]: This method should be async.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java [hunk 3]
```diff
@@ -1373,43 +1373,33 @@ protected DispatchRate internalGetSubscriptionDispatchRate() {
...
-        }
+        });
     }
 
-    protected void internalDeleteSubscribeRate() {
-        validateSuperUserAccess();
-        try {
+    protected CompletableFuture<Void> internalDeleteSubscribeRateAsync() {
+        return validateSuperUserAccessAsync().thenAccept(__ -> {
             updatePolicies(namespaceName, policies -> {
```
Comment (mattisonchao) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:1390]: This method should be async.

**Review by mattisonchao — COMMENTED** [review]
Great work!  I left some comments here, could you please help confirm the test cover this change?

**Comment by gaoyanliang** [comment]
@mattisonchao OK, I'll try it. There is no unit test for the v2 Namespaces yet, I will try to add it, it may take some time.

**Comment by gaoyanliang** [comment]
image

Do these non-existent files need to be generated locally？

**Comment by gaoyanliang** [comment]
@shibd  @mattisonchao  PTAL.

**Comment by gaoyanliang** [comment]
Hi @mattisonchao Can you help trigger the ci ？

**Review by shibd — APPROVED** [review]
/LGTM

**Review by Technoboy- — APPROVED** [review]

**Comment by gaoyanliang** [comment]
Hi @Technoboy-  Can you help trigger the ci ？

**Comment by Technoboy-** [comment]
> Hi @Technoboy- Can you help trigger the ci ？

Yes, I have triggered.

**Comment by gaoyanliang** [comment]
> Yes, I have triggered.

Thanks

**Pull request merged**